src/pfs/fuse/fuse.go

```diff
@@ -0,0 +1,127 @@
+package fuse
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"bazil.org/fuse"
+	"bazil.org/fuse/fs"
+	"github.com/pachyderm/pachyderm/src/pfs"
+	"github.com/pachyderm/pachyderm/src/pfs/pfsutil"
+	"golang.org/x/net/context"
+)
+
+func Mount(apiClient pfs.ApiClient, repositoryName string, commitID string, mountPoint string) error {
+	if err := os.MkdirAll(mountPoint, 0777); err != nil {
+		return err
+	}
+	c, err := fuse.Mount(
+		mountPoint,
+		fuse.FSName("pfs"),
+		fuse.Subtype("pfs"),
+		fuse.VolumeName("pfs://"+repositoryName),
+	)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	if err := fs.Serve(c, &filesystem{apiClient, repositoryName, commitID}); err != nil {
+		return err
+	}
+
+	// check if the mount process has an error to report
+	<-c.Ready
+	return c.MountError
+}
+
+func Unmount(mountPoint string) error {
+	return nil
+}
+
+type filesystem struct {
```

Comment:
with the rest of the code in src/pfs, the style I have is:

```
src/path/to/something/something.go # all public methods, nothing else
src/path/to/something/filesystem.go # filesystem struct, private constructor, methods
```

We can talk about it but this works REALLY well especially with onboarding new people into the codebase